File: algorithms/1779.find-nearest-point-that-has-the-same-x-or-y-coordinate/solution.rs
impl Solution {
    pub fn nearest_valid_point(x: i32, y: i32, points: Vec<Vec<i32>>) -> i32 {
        let mut dists: Vec<Vec<i32>> = vec![];

        for (idx, point) in points.iter().enumerate() {
            let (px, py) = (point[0], point[1]);
            if x == px || y == py {
                dists.push(vec![(x - px).abs() + (y - py).abs(), idx as i32])
            }
        }

        if dists.len() == 0 {
            -1
        } else {
            dists.sort();
            dists[0][1]
        }
    }
}

File: algorithms/1779.find-nearest-point-that-has-the-same-x-or-y-coordinate/solution.rb
# @param {Integer} x
# @param {Integer} y
# @param {Integer[][]} points
# @return {Integer}
def nearest_valid_point(x, y, points)
  dists = []

  points.each_with_index do |point, idx|
    px, py = point
    if x == px || y == py
      dists << [(x - px).abs + (y - py).abs, idx]
    end
  end

  dists.empty? ? -1 : dists.sort[0][1]
end
